xdsl/passes.py

```diff
@@ -1,20 +1,123 @@
 from abc import ABC, abstractmethod
+from dataclasses import dataclass, Field
+from types import UnionType, NoneType
+
+from xdsl.utils.hints import isa
 from xdsl.dialects import builtin
 from xdsl.ir import MLContext
-from typing import ClassVar
+from typing import ClassVar, TypeVar, get_origin, Any, Union, get_args
+from xdsl.utils.parse_pipeline import (
+    PipelinePassSpec,
+    PassArgListType,
+    PassArgElementType,
+)
+
+_T = TypeVar("_T", bound="ModulePass")
```

Comment:
```suggestion
ModulePassInvT = TypeVar("ModulePassInvT", bound="ModulePass")
```
To remain consistent with `ir.py` `TypeVar`?

Reply:
I added a type that is neither inv or cov, I hope that's correct^^